embark/workers/orchestrator.py

```diff
@@ -126,6 +126,18 @@ def assign_worker(self, worker: Worker, task: OrchestratorTask):
         with self.lock:
             self._assign_worker(worker, task)
 
+    def free_worker(self, worker: Worker):
+        """
+        Marks a busy worker as free.
+
+        :param worker: The worker to be marked free
+        """
+        with self.lock:
+            del self.busy_workers[worker.ip_address]
+            self.free_workers[worker.ip_address] = worker
+            worker.analysis_id = None
+            worker.save()
```

Comment:
If possible, please replace this function with `release_worker` as suggested and delete the definition for `release_worker` below.
The `release_worker` definition that assigns another task right after releasing a worker is obsolete after the merge of Issue 59. 

```suggestion
    def release_worker(self, worker: Worker):
        """
        Release a busy worker from its current task and mark it as free.

        :param worker: The worker to be released
        :raises ValueError: If the worker is not busy
        """
        with self.lock:
            if worker.ip_address not in self.busy_workers:
                raise ValueError(f"Worker with IP {worker.ip_address} is not busy.")

            del self.busy_workers[worker.ip_address]
            self.free_workers[worker.ip_address] = worker
            worker.analysis_id = None
            worker.save()
```